### PR TITLE
Add a hook that surfaces the regression hygiene checklist automatically

### PR DESCRIPTION
**Pull request by jkomg**
## Summary
Following the full codebase architecture crawl (which surfaced several latent bugs — see PR #363/#364), adds a `PreToolUse` hook on Edit/Write so the regression hygiene checklist gets automatically injected into context on the first code edit each day, rather than relying on remembering to check memory.

## What it does
- Fires on `Edit`/`Write`, deduped per calendar day via a `/tmp` marker (not per-session — session IDs resolve to an empty string in this environment, which would have made a session-based dedup fire only once ever until `/tmp` is cleared; caught and fixed during testing before finalizing).
- Injects the full content of `feedback_regression_hygiene_checklist.md` (from Claude's memory directory) plus a pointer to the architecture-map memory files for deeper lookups on shared models/sync pairing/worker footguns.
- No-ops silently (`{}`) on every edit after the first each day — not spammy.

## Test plan
- [x] Pipe-tested the raw command against synthetic hook input matching the real schema (including the empty-`session_id` edge case)
- [x] Validated JSON syntax and hook schema shape (`jq -e` against the settings file)
- [x] Proved it fires live in-session (triggered a real Edit, confirmed the full checklist appeared as injected context; confirmed the second edit in the same run did not re-fire)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/settings.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "hooks": {
+    "PreToolUse": [
+      {
+        "matcher": "Edit|Write",
+        "hooks": [
+          {
+            "type": "command",
+            "command": "marker=\"/tmp/claude-hygiene-hook-$(date +%Y%m%d)\"; if [ ! -f \"$marker\" ]; then touch \"$marker\"; checklist=$(cat \"/Users/jkennedy/.claude/projects/-Users-jkennedy-Projects-mcbn-xp-tracker/memory/feedback_regression_hygiene_checklist.md\" 2>/dev/null); ctx=\"Regression hygiene reminder (first code edit today) -- full checklist below. Also consult project_architecture_web.md / project_architecture_bot.md / project_history_and_themes.md in the same memory dir for deeper context (shared models, Sheets/audit-log sync pairing, the worker enabled-flag footgun table) before touching db.py, db_service.py, sheets.py/sheets_sync.py, or apps/bot/src/services.\\n\\n${checklist}\"; jq -n --arg ctx \"$ctx\" '{\"hookSpecificOutput\":{\"hookEventName\":\"PreToolUse\",\"additionalContext\":$ctx}}'; else echo '{}'; fi",
```
[the remark this answers is not in the document]
Comment (jkomg) [reply .claude/settings.json]: Fixed in e5b0329 — you're right, the checklist source lived only in Claude's private per-machine memory directory, an absolute path under one developer's home dir. Committed the actual content as real docs (docs/ARCHITECTURE_WEB.md, ARCHITECTURE_BOT.md, PROJECT_HISTORY_AND_THEMES.md, REGRESSION_HYGIENE_CHECKLIST.md) so they're portable to any checkout, and the hook now resolves the path via `git rev-parse --show-toplevel` instead of a hardcoded absolute path. Also fixed the silent-failure issue you flagged: the /tmp marker is now only touched after confirming the file was actually read (non-empty), so a broken path retries on the next edit instead of permanently suppressing itself for the rest of the day. Pipe-tested all three cases (success, same-day dedup, missing-file no-op) before committing.
